lib/flop.ex

```diff
@@ -1959,17 +1975,24 @@ defmodule Flop do
       iex> push_order(%Flop{order_by: [:id], before: "DEF"}, :name)
       %Flop{order_by: [:name, :id], order_directions: [:asc], before: nil}
   """
-  @spec push_order(Flop.t(), atom | String.t()) :: Flop.t()
+  @spec push_order(Flop.t(), atom | String.t(), [option()]) :: Flop.t()
   @doc since: "0.10.0"
   @doc group: :parameters
+  def push_order(flop, field, opts \\ [])
+
   def push_order(
         %Flop{order_by: order_by, order_directions: order_directions} = flop,
-        field
+        field,
+        opts
       )
       when is_atom(field) do
     previous_index = get_index(order_by, field)
     previous_direction = get_order_direction(order_directions, previous_index)
-    new_direction = new_order_direction(previous_index, previous_direction)
+
+    directions = Keyword.get(opts, :directions, nil)
```

Comment:
I think it would be easier to use a tuple as default:

```suggestion
    directions = Keyword.get(opts, :directions, {:asc, :desc})
```

Then you don't have to handle the `nil` case below.

Reply:
You rely on being able to pass `nil` in the flop_phoenix PR, but I'd rather default to `{:asc, :desc}` or just pass an empty list as `opts` if the attribute is not set by the user.

Reply:
Made the upstream change in `flop_phoenix` in https://github.com/woylie/flop_phoenix/pull/242/commits/9fbc1d28e00fd6a00c3828544d8b1d4602334c69 and it blends as expected.

Reply:
Yes, you're right.